coalib/output/ConsoleInteraction.py

```diff
@@ -164,35 +158,48 @@ def print_lines(console_printer,
     :param sourcerange:     The SourceRange object referring to the related
                             lines to print.
     """
+
     for i in range(sourcerange.start.line, sourcerange.end.line + 1):
         console_printer.print(format_lines(lines='', line_nr=i),
                               color=FILE_LINES_COLOR,
                               end='')
+
         line = file_dict[sourcerange.file][i - 1].rstrip("\n")
-        tab_width = int(section.get('tab_width', 4))
-        s = SpacingHelper(tab_width)
-        tab_dict = dict(s.yield_tab_lengths(line))
+        # Get suitable lexer depending on filename's extension and if not found
+        # fallback to TextLexer().
+        try:
+            lexer = get_lexer_for_filename(sourcerange.file)
+        except ClassNotFound:
+            lexer = TextLexer()
+        lexer.add_filter(VisibleWhitespaceFilter(
+                             spaces="•", tabs=True,
+                             tabsize=SpacingHelper.DEFAULT_TAB_WIDTH))
+        # highlight() combines lexer and formatter to output a ``str``
+        # object.
         printed_chars = 0
         if i == sourcerange.start.line and sourcerange.start.column:
-            print_spaces_tabs_in_unicode(
-                console_printer, line[:sourcerange.start.column-1],
-                tab_dict, FILE_LINES_COLOR)
+            console_printer.print(highlight(
+                line[:sourcerange.start.column-1],
+                lexer, TerminalTrueColorFormatter())[:-1], end='')
 
             printed_chars = sourcerange.start.column-1
 
         if i == sourcerange.end.line and sourcerange.end.column:
-            print_spaces_tabs_in_unicode(
-                console_printer, line[printed_chars:sourcerange.end.column-1],
-                tab_dict, HIGHLIGHTED_CODE_COLOR, printed_chars)
-
-            print_spaces_tabs_in_unicode(
-                console_printer, line[sourcerange.end.column-1:],
-                tab_dict, FILE_LINES_COLOR, sourcerange.end.column-1)
+            console_printer.print(highlight(
+                line[printed_chars:sourcerange.end.column-1],
+                lexer, TerminalTrueColorFormatter(
+                    style=BackgroundSourceRangeStyle))[:-1], end='')
+
+            console_printer.print(highlight(
+                line[sourcerange.end.column-1:],
+                lexer, TerminalTrueColorFormatter())[:-1], end='')
             console_printer.print("")
+
         else:
-            print_spaces_tabs_in_unicode(
-                console_printer, line[printed_chars:], tab_dict,
-                HIGHLIGHTED_CODE_COLOR, printed_chars)
+            console_printer.print(highlight(
+                line[printed_chars:], lexer,
+                TerminalTrueColorFormatter(
+                    style=BackgroundSourceRangeStyle))[:-1], end='')
```

Comment:
Hm so what I feel is that it checks for that language's special reserved keywords like `while` `in` `for` `range` `if` `else` and stuff like that. Also distinguish on data types etc. That works well even if it's one line at a time. If there are other cases, then that would be a limitation of Pygments as well, I believe. But I feel syntax highlighting presently works well :+1: 

I did a small experiment in python shell by putting the string at once, to see if there's a difference but the result is pretty same and (expected) too.
in coala
![screenshot from 2016-08-10 14-47-32](https://cloud.githubusercontent.com/assets/5689132/17548429/82995700-5f09-11e6-88fe-a0e0818f78b6.png)

in python shell
 - Only one line supplied.
![screenshot from 2016-08-10 14-53-43](https://cloud.githubusercontent.com/assets/5689132/17548609/41b11c36-5f0a-11e6-8075-bd22db2a2f63.png)

 - Full file supplied
![screenshot from 2016-08-10 14-52-33](https://cloud.githubusercontent.com/assets/5689132/17548576/17bc995a-5f0a-11e6-881e-79d631bbf2da.png)